### ISSUE TITLE
Bug: Issue using Shared Libraries when I use drive_id (Key not found: remaining)

### ISSUE THREAD

**Issue by pedro-aguia** [abraunegg/onedrive#2588]
### Describe the bug

As soon as I put something in the drive_id field the json exceptions begins

### Operating System Details

```shell
Linux 22.04
```

### Client Installation Method

From Source

### OneDrive Account Type

Business | Office365

### What is your OneDrive Application Version

onedrive v2.4.13-1build1

### What is your OneDrive Application Configuration

```shell
Configuration file successfully loaded
onedrive version                       = v2.4.13-1build1
Config path                            = /home/xx/.config/Datos
Config file found in config path       = true
Config option 'check_nosync'           = false
Config option 'sync_dir'               = /home/xx/Datos
Config option 'skip_dir'               =
Config option 'skip_file'              = ~*|.~*|*.tmp
Config option 'skip_dotfiles'          = false
Config option 'skip_symlinks'          = false
Config option 'monitor_interval'       = 300
Config option 'min_notify_changes'     = 5
Config option 'log_dir'                = /var/log/onedrive/
Config option 'classify_as_big_delete' = 1000
Config option 'upload_only'            = false
Config option 'no_remote_delete'       = false
Config option 'remove_source_files'    = false
Config option 'drive_id'               = b!xxxxxxxxxxx
Config option 'sync_root_files'        = false
Selective sync 'sync_list' configured  = false
Business Shared Folders configured     = false
```

### What is your 'curl' version

```shell
curl 7.81.0 (x86_64-pc-linux-gnu) libcurl/7.81.0 OpenSSL/3.0.2 zlib/1.2.11 brotli/1.0.9 zstd/1.4.8 libidn2/2.3.2 libpsl/0.21.0 (+libidn2/2.3.2) libssh/0.9.6/openssl/zlib nghttp2/1.43.0 librtmp/2.3 OpenLDAP/2.5.14
Release-Date: 2022-01-05
Protocols: dict file ftp ftps gopher gophers http https imap imaps ldap ldaps mqtt pop3 pop3s rtmp rtsp scp sftp smb smbs smtp smtps telnet tftp
Features: alt-svc AsynchDNS brotli GSS-API HSTS HTTP2 HTTPS-proxy IDN IPv6 Kerberos Largefile libz NTLM NTLM_WB PSL SPNEGO SSL TLS-SRP UnixSockets zstd
```

### Where is your 'sync_dir' located

Local

### What are all your system 'mount points'

```shell
sysfs on /sys type sysfs (rw,nosuid,nodev,noexec,relatime)
proc on /proc type proc (rw,nosuid,nodev,noexec,relatime)
udev on /dev type devtmpfs (rw,nosuid,relatime,size=32761600k,nr_inodes=8190400,mode=755,inode64)
devpts on /dev/pts type devpts (rw,nosuid,noexec,relatime,gid=5,mode=620,ptmxmode=000)
tmpfs on /run type tmpfs (rw,nosuid,nodev,noexec,relatime,size=6559952k,mode=755,inode64)
/dev/nvme0n1p6 on / type ext4 (rw,relatime,errors=remount-ro)
securityfs on /sys/kernel/security type securityfs (rw,nosuid,nodev,noexec,relatime)
tmpfs on /dev/shm type tmpfs (rw,nosuid,nodev,inode64)
tmpfs on /run/lock type tmpfs (rw,nosuid,nodev,noexec,relatime,size=5120k,inode64)
cgroup2 on /sys/fs/cgroup type cgroup2 (rw,nosuid,nodev,noexec,relatime,nsdelegate,memory_recursiveprot)
pstore on /sys/fs/pstore type pstore (rw,nosuid,nodev,noexec,relatime)
efivarfs on /sys/firmware/efi/efivars type efivarfs (rw,nosuid,nodev,noexec,relatime)
bpf on /sys/fs/bpf type bpf (rw,nosuid,nodev,noexec,relatime,mode=700)
systemd-1 on /proc/sys/fs/binfmt_misc type autofs (rw,relatime,fd=29,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=33316)
hugetlbfs on /dev/hugepages type hugetlbfs (rw,relatime,pagesize=2M)
mqueue on /dev/mqueue type mqueue (rw,nosuid,nodev,noexec,relatime)
debugfs on /sys/kernel/debug type debugfs (rw,nosuid,nodev,noexec,relatime)
tracefs on /sys/kernel/tracing type tracefs (rw,nosuid,nodev,noexec,relatime)
fusectl on /sys/fs/fuse/connections type fusectl (rw,nosuid,nodev,noexec,relatime)
configfs on /sys/kernel/config type configfs (rw,nosuid,nodev,noexec,relatime)
ramfs on /run/credentials/systemd-sysusers.service type ramfs (ro,nosuid,nodev,noexec,relatime,mode=700)
/dev/nvme0n1p2 on /boot/efi type vfat (rw,relatime,fmask=0077,dmask=0077,codepage=437,iocharset=iso8859-1,shortname=mixed,errors=remount-ro)
binfmt_misc on /proc/sys/fs/binfmt_misc type binfmt_misc (rw,nosuid,nodev,noexec,relatime)
/dev/sda2 on /media/data type fuseblk (rw,nosuid,nodev,relatime,user_id=0,group_id=0,allow_other,blksize=4096,x-gvfs-show)
tmpfs on /run/user/1001 type tmpfs (rw,nosuid,nodev,relatime,size=6559948k,nr_inodes=1639987,mode=700,uid=1001,gid=1001,inode64)
gvfsd-fuse on /run/user/1001/gvfs type fuse.gvfsd-fuse (rw,nosuid,nodev,relatime,user_id=1001,group_id=1001)
portal on /run/user/1001/doc type fuse.portal (rw,nosuid,nodev,relatime,user_id=1001,group_id=1001)
tmpfs on /run/user/1000 type tmpfs (rw,nosuid,nodev,relatime,size=6559948k,nr_inodes=1639987,mode=700,uid=1000,gid=1000,inode64)
gvfsd-fuse on /run/user/1000/gvfs type fuse.gvfsd-fuse (rw,nosuid,nodev,relatime,user_id=1000,group_id=1000)
portal on /run/user/1000/doc type fuse.portal (rw,nosuid,nodev,relatime,user_id=1000,group_id=1000)
/dev/nvme0n1p3 on /media/windows type fuseblk (rw,relatime,user_id=0,group_id=0,allow_other,blksize=4096)
```

### What are all your local file system partition types

```shell
NAME        FSTYPE FSVER LABEL UUID                                 FSAVAIL FSUSE% MOUNTPOINTS
sda
├─sda1
└─sda2      ntfs         DATA  C28xxxxx62xxxxD                        3,4T     6% /media/data
nvme0n1
├─nvme0n1p1
├─nvme0n1p2 vfat   FAT32       B6xxxxx74                              65,9M    34% /boot/efi
├─nvme0n1p3 ntfs         OS    DAxxxxxD0F3700FC                      223,2G    50% /media/windows
├─nvme0n1p4 ntfs               FBFxxxxxF13D9xxB
├─nvme0n1p5 swap   1           15xxxxd0-ec29-4d8c-aa05-7bxxxxxadb95                [SWAP]
└─nvme0n1p6 ext4   1.0         8dxxxxx9-f6c1-457-afa9-e97xxxxx803da  146,6G    60% /
```

### How do you use 'onedrive'

The idea is to sync one SharePoint Site with all the folders it has on it, like we can do on windows (something from sharepoint.com/sites/xxxxxxx)

### Steps to reproduce the behaviour

I followed the documentation (https://github.com/abraunegg/onedrive/blob/master/docs/SharePoint-Shared-Libraries.md) step by step

### Complete Verbose Log Output

```shell
Using 'user' Config Dir: /home/aguia/.config/Datos
Using 'system' Config Dir:
Configuration file successfully loaded
DRY-RUN Configured. Output below shows what 'would' have occurred.
Initializing the OneDrive API ...
Configuring Global Azure AD Endpoints
Authorize this app visiting:

https://login.microsoftonline.com/common/oauth2/v2.0/authorize?client_id=d50ca740-c83f-4d1b-b616-

Enter the response uri: https://login.m

Opening the item database ...
All operations will be performed in: /home/aguia/Datos
std.json.JSONException@std/json.d(641): Key not found: remaining
----------------
??:? pure @safe noreturn std.exception.bailOut!(std.json.JSONException).bailOut(immutable(char)[], ulong, scope const(char)[]) [0x7ffa9591d9e2]
??:? inout pure ref @safe inout(std.json.JSONValue) std.json.JSONValue.opIndex(return immutable(char)[]) [0x7ffa95895169]
??:? [0x558268c27f25]
??:? [0x558268c6c2b4]
??:? void rt.dmain2._d_run_main2(char[][], ulong, extern (C) int function(char[][])*).runAll() [0x7ffa9559ea4b]
??:? _d_run_main2 [0x7ffa9559e867]
??:? _d_run_main [0x7ffa9559e6bd]
??:? [0x7ffa95029d8f]
??:? __libc_start_main [0x7ffa95029e3f]
??:? [0x558268c171c4]
std.net.curl.CurlException@std/net/curl.d(4402): A libcurl function was given a bad argument on handle null
----------------
??:? pure @safe noreturn std.exception.bailOut!(std.net.curl.CurlException).bailOut(immutable(char)[], ulong, scope const(char)[]) [0x7ffa9591dad7]
??:? void std.net.curl.Curl.clear(etc.c.curl.CurlOption) [0x7ffa9587e363]
??:? [0x558268c673e8]
??:? void rt.dmain2._d_run_main2(char[][], ulong, extern (C) int function(char[][])*).runAll() [0x7ffa9559ea4b]
??:? _d_run_main2 [0x7ffa9559e867]
??:? _d_run_main [0x7ffa9559e6bd]
??:? [0x7ffa95029d8f]
??:? __libc_start_main [0x7ffa95029e3f]
??:? [0x558268c171c4]
```

**Comment by pedro-aguia** [comment]
Solved installing version: **onedrive v2.4.25-1+np1**

**Comment by abraunegg** [comment]
> Solved installing version: **onedrive v2.4.25-1+np1**

Correct - as per the documentation you must be using a supported client version. Ubuntu and it's clones have unsupported packages in the Universe repository that contain bugs which have been fixed - which is why you must use the OpenSuSE repository to install.

**Comment by abraunegg** [comment]
This issue has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.
